src/main/java/seedu/address/model/matric/Matric.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.matric;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Matriculation Number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMatric(String)}
+ */
+public class Matric {
+    public static final String MESSAGE_CONSTRAINTS = "Matriculation numbers in the style of A1234567Z are accepted."
+            + "The first character must be an uppercase letter,"
+            + "followed by 7 digits and ending with an uppercase letter.";
+    public static final String VALIDATION_REGEX = "[A-Z][0-9]{7}[A-Z]";
```

Comment:
Thank you Johan. I agree we could get more inputs on this matter but for now I choose to be cautious and allow A-Z for the first character, what do the others think?

Reply:
I think if we are limiting it for NUS head tutors, we can set it to start with A. This can prevent erroneous inputs / can warn the user of any erroneous inputs.

Reply:
Nice! I think however that this should be implemented in later versions. For now we can focus on finishing our deliverables